### DISCUSSION TITLE
UTM can't start a virtual mips machine

### DISCUSSION THREAD

**Discussion by 752228595** [utmapp/UTM#2417] [UTM Help (macOS)]
I have downloaded a iso (debian-10.8.0-mips64el-xfce-CD-1.iso), and plan to install it with UTM. But it won't work. When I build it with mips64(little endian) and MIPS Malta Core LV, it just black screen , nothing shows up in the application window.(no matter I load the iso or not, it still the same. I think it just can't boot up)

If it needs a special way to run the special arch，please inform me. Thankyou.

PS: I have run iso with x86_64 and aarch64 successfully. And I run mips64el at the same way, but it won't work.(of course I have selected with different arch and cpu model)

**Comment by conath** [comment]
Hi @752228595, thanks for the detailed explanation. However, I can't reproduce this on macOS - instead I get a crash. Are you on Mac or iOS?

I tried running a blank MPS64 (little endian) machine on my Mac (macOS 11.2.2) in UTM (2.0.23). This results in a crash when starting the VM:

`BUG IN CLIENT OF LIBPLATFORM: Trying to recursively lock an os_unfair_lock` in `libsystem_platform.dylib`.

VM is attached.

[MIPS64.utm.zip](https://github.com/utmapp/UTM/files/6056852/MIPS64.utm.zip)

I am unfamiliar with this type of system, so not sure if e. g. a firmware file needs to be supplied to make it work.

**Comment by 752228595** [comment]
Hi @conath , thanks for replying.
I am using utm on Mac with apple silicon. And I have tested your vm just now, it still the same --- black screen.
About the firmware, I think it might need it. 
Here is the url , I have tested successfully in qemu. Hope it would help.
https://markuta.com/how-to-build-a-mips-qemu-image-on-debian/

**Comment by conath** [comment]
Ok, here's how you "translate" the commands to UTM configuration. You need to import the kernel and ramdisk image files as Drives and then change the Image Type.

<img width="1330" alt="Bildschirmfoto 2021-02-28 um 15 08 51" src="https://user-images.githubusercontent.com/12073163/109421367-46577b80-79d7-11eb-900e-6fead2029dd8.png">

Please let me know if it works.

**Comment by 752228595** [comment]
No, it still won't work. Still black screen.

**Comment by conath** [comment]
Are you trying to get a Desktop OS running or console mode? In case of console only, you will need to switch to "Console Only" in the Graphics configuration.

**Comment by 752228595** [comment]
When i run it with qemu, it doesn't have gui.(Console mode)
And I have tried "Console Only" mode in UTM. But it still won't work, nothing shows up (black screen).(Different : A small red vertical line is in the upper left corner.)

**Comment by osy** [comment]
Going by the instructions listed, make sure you have "malta" selected as your system target. Then make sure you have the drives set up as Linux Kernel and Linux RAM Disk. Finally, in QEMU tab add `append "root=/dev/sda1 console=ttyS0 nokaslr"` as an additional argument on the bottom. This should generate a QEMU argument that is the same as the one in the guide (with other added arguments). You can also debug by exporting the arguments to get the full QEMU command and running it.

MIPS/other architectures are not fully supported yet so they require a bit more work to get running. Right now the focus is on x86, arm, ppc which are the ones most people are using.
